spot-client/src/common/backend/SpotBackendService.test.js

```diff
@@ -210,6 +236,43 @@ describe('SpotBackendService', () => {
                             });
                         });
                     }));
+
+            it('will not emit registration lost when 401 is returned to get room info', () => {
+                const registrationLostCallback = jest.fn();
+
+                spotBackendService.addListener(
+                    SpotBackendService.REGISTRATION_LOST,
+                    registrationLostCallback
+                );
+
+                return spotBackendService.register(MOCK_PAIRING_CODE)
+                    .then(() => {
+                        // Mock the initial request to get room info
+                        fetch.once('', {
+                            status: 401,
+                            ok: false
+                        });
+
+                        // Mock the refreshing of the token
+                        fetch.once(JSON.stringify(MOCK_RESPONSE));
+
+                        // Mock the request to get room info
+                        fetch.once(JSON.stringify({
+                            id: 'mock-id',
+                            mucUrl: 'muc-url',
+                            name: 'mock-muc-name'
+                        }));
+
+                        return spotBackendService.getRoomInfo().then(roomInfo => {
+                            expect(roomInfo).toEqual({
+                                id: 'mock-id',
+                                name: 'mock-muc-name',
+                                roomName: 'muc-url'
+                            });
```

Comment:
I see a chunk of duplication with the above. Do you think it's worthwhile to move this into a helper?

Reply:
right, will do

Reply:
Okay I gave it a try, please let me know if that's what you had in mind. It's in separate commit and eventually could be left this way or I can squash and rebase if you want me to.